### DISCUSSION TITLE
Why `addChildren` and `getParentRoute` are both required?

### DISCUSSION THREAD

**Discussion by DaniilDenisyuk** [TanStack/router#585] [Q&A] [answered]
If `getParentRoute` returns parent of route being created, then why not to automatically set created route as a child of its parent?
Logically seems to be correct. Also a plus for code splitting as this removes one place to change when you want to restructure your route tree

**Comment by bhaugeea** [comment, marked as answer]
I think this is more for TypeScript typing inference than for runtime behavior.

**Reply by tannerlinsley** [reply]
Correct. Without this, TS would have no idea about the relationships between your parent/child routes.

**Reply by vcarel** [reply]
Is it possible to have a helper to create the routes all at once? the v4 way is a bit complicated.

Since I'm lazy, I asked ChatGPT to provide an example 😅
It actually makes sense, but would the types still work?

```ts
import { RootRoute, Route } from '@tanstack/router';

interface RouteDefinition {
  path: string;
  parent?: string;
  children?: RouteDefinition[];
}

function createRoutes(routeDefinitions: RouteDefinition[], parent?: Route): Route[] {
  const getParentRoute = parent ? () => parent : undefined;
  return routeDefinitions.map(({ path, children }) => {
    const route = new Route({ getParentRoute, path });
    if (children) {
      route.addChildren(createRoutes(children, route));
    }
    return route;
  });
}

// Usage
const routeDefinitions: RouteDefinition[] = [
  { path: '/' },
  {
    path: 'blog',
    children: [{ path: '$slug' }],
  },
];

const rootRoute = new RootRoute();
const routes = createRoutes(routeDefinitions);
const routeTree = rootRoute.addChildren(routes);
```
